### PR TITLE
Refactoring package refresh

### PR DESCRIPTION
**Pull request by EnKarin**

### CODE REVIEW

File: src/main/java/io/github/enkarin/bookcrossing/refresh/controllers/RefreshController.java [hunk 1]
```diff
@@ -29,33 +28,38 @@
...
-        final ErrorListResponse response = new ErrorListResponse();
-        response.getErrors().add("refresh: Неверный или истекший токен");
-        return new ResponseEntity<>(response, HttpStatus.FORBIDDEN);
+        return ResponseEntity.ok(refreshService.createTokens(request.getRefresh()));
+    }
+
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    @ExceptionHandler(TokenNotFoundException.class)
+    public Map<String, String> tokenNotFound(final TokenNotFoundException exc) {
+        return Collections.singletonMap("refresh", exc.getMessage());
```
Comment (mfvanek) [inline src/main/java/io/github/enkarin/bookcrossing/refresh/controllers/RefreshController.java:57]: Use modern API such as Map.of

File: src/main/java/io/github/enkarin/bookcrossing/refresh/service/RefreshService.java [hunk 2]
```diff
@@ -1,27 +1,45 @@
...
 
-    public String createToken(final String login) {
+    public AuthResponse createTokens(final String token) {
+        final Refresh refresh = refreshRepository.findById(token).orElseThrow(TokenNotFoundException::new);
+        if (LocalDate.ofEpochDay(refresh.getDate()).isAfter(LocalDate.now())) {
+            return AuthResponse.create(jwtProvider.generateToken(refresh.getUser()),
+                    createAccess(refresh.getUser()));
+        } else {
+            refreshRepository.delete(refresh);
+            throw new RefreshTokenInvalidException();
```
Comment (mfvanek) [inline src/main/java/io/github/enkarin/bookcrossing/refresh/service/RefreshService.java:38]: You have to create a test for this method. Throwing RefreshTokenInvalidException will lead to transaction rollback thus no data will be deleted from the database

**Review by mfvanek — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/enkarin/bookcrossing/refresh/service/RefreshService.java (hunk 2 above)
Comment (mfvanek) [reply src/main/java/io/github/enkarin/bookcrossing/refresh/service/RefreshService.java:38]: Take a look at noRollbackFor https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/transaction/annotation/Transactional.html

File: src/main/java/io/github/enkarin/bookcrossing/refresh/controllers/RefreshController.java (hunk 1 above)
Comment (EnKarin) [reply src/main/java/io/github/enkarin/bookcrossing/refresh/controllers/RefreshController.java]: Changed

File: src/main/java/io/github/enkarin/bookcrossing/refresh/service/RefreshService.java (hunk 2 above)
Comment (EnKarin) [reply src/main/java/io/github/enkarin/bookcrossing/refresh/service/RefreshService.java:38]: Added

**Review by mfvanek — APPROVED** [review]

**Pull request merged**